### PR TITLE
Hyperapp official website

### PR DESCRIPTION
**Pull request by loteoo**
Hello everyone!

So here it is! The upcoming new website and documentation for Hyperapp. 🎉

It's the result of multiple iterations of designs and implementations from @jorgebucaran and myself.

@jorgebucaran  designed a beautiful **retro yet *fresh*** website and I very slowly implemented it over the past few months.

I can describe the design as a brutalistic approach while keeping a clean look.

We've been working on this for a while now and are ready to get more eyes on the project. 🙏  

The website will be hosted on Github pages here: https://hyperapp.dev/

You can see a development preview here: https://hyperapp-beta.netlify.com/

To build the site, you can run `npm run build` inside the `/web` directory, which will build a parcel bundle in the `/docs` directory for Github pages to use.

Feel free to discus ideas, suggestions and improvements of the site here.

You can even do a PR to this PR by forking [my fork](https://github.com/loteoo/hyperapp/tree/feature/hyperapp-official-site)!

Merges on the [fork](https://github.com/loteoo/hyperapp/tree/feature/hyperapp-official-site) will automatically be deployed on the [development](https://hyperapp-beta.netlify.com/) website.

You can also just let me know about possible improvements and me or @jorgebucaran will have a look at it.

I think the idea is to get a first version out soon, and iterate on it over time.

Please, let us know what you think!! 💯 

hyperapp-beta netlify com_(iPad Pro)

hyperapp-beta netlify com_(iPad)

hyperapp-beta netlify com_(iPhone 6_7_8)

### CODE REVIEW

**Comment by jorgebucaran** [comment]
@loteoo `docs_backup`?

File: web/src/subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    dispatch([props.action, { ev, scrollY: window.scrollY }])
+  }
+  addEventListener('scroll', handleScroll)
+  return () => {
+    removeEventListener('scroll', handleScroll)
+  }
+})
+
+// PopState Subscription
+const subFx = a => b => [a, b]
```
Comment (SkipKayhil) [inline web/src/subscriptions.js]: I believe this is the same function as on line 3, you should be able to use it for both subs

Comment (loteoo) [reply web/src/subscriptions.js]: Nice catch, will fix

**Comment by loteoo** [comment]
> @loteoo `docs_backup`?

Haha I wasn't sure what to do with the previous site, it's now deleted. 🤷‍♂ 

Now will I have to bring back the tutorial in this PR?

**Comment by vuongnhfsoft** [comment]
Syntax highlighting is not working properly: if I navigate away from the homepage (to tutorial page for example) and then navigate back, syntax highlighting is lost.

Also the page currently loads 4 web fonts, taking up 80-90% of download size. Perhaps we can shed some weight by cutting 2 of them, leaving one for headline and one for code.

**Comment by jorgebucaran** [comment]
> Also the page currently loads 4 web fonts, taking up 80-90% of download size. Perhaps we can shed some weight by cutting 2 of them, leaving one for headline and one for code.

@vuongnhfsoft How to do that without changing the design of the site?

> @loteoo Now will I have to bring back the tutorial in this PR?

Yes, please.

**Comment by vuongnhfsoft** [comment]
@jorgebucaran there's no way. The design will definitely change, the question is how much.

For syntax highlighting: currently the page downloads 2 FiraCode fonts for 2 font weights. FiraCode Bold is only used for keywords like `import`. If we use different color instead of different weight, we can omit FiraCode Bold.

For body text: perhaps Github font stack for replacement.
`font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;`

Current body text:
current-body

Github font stack on my Windows machine (resolves to Segoe UI most likely). I don't have access to other systems right now.
new-body

**Comment by jorgebucaran** [comment]
> @vuongnhfsoft For syntax highlighting: currently the page downloads 2 FiraCode fonts for 2 font weights. FiraCode Bold is only used for keywords like import. If we use different color instead of different weight, we can omit FiraCode Bold.

This one works for me. 👌

**Comment by loteoo** [comment]
@vuongnhfsoft Thank you for your suggestions, I have applied them and we are now down to 2 fonts. I made the highlight apply after navigations as well.

@jorgebucaran I switched the highlighting library to prismjs as you requested.

Let me know of any more tweaks to do!

File: web/src/pages/Home/counterCode.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export default `import { h, app } from "hyperapp"
+
+app({
+  init: 0,
+  view: state =>
+    h("div", {}, [
+      h("h1", {}, state),
+      h("button", { onclick: state => state - 1 }, "-"),
```
Comment (okwolf) [inline web/src/pages/Home/counterCode.js]: The code uses "-" and "+", but the live example has buttons for "subtract" and "add".

**Comment by jorgebucaran** [comment]
@loteoo Can you resolve conflicts? Let's merge this. 🙏

**Comment by loteoo** [comment]
@jorgebucaran Just rebased, conflict resolved!

Do you want me to commit the built site to the PR before we merge? (result of running `npm run build-site`), or will you have some sort of CI/CD for this?

**Comment by jorgebucaran** [comment]
@loteoo So, essentially what I thought was that merging this would automatically deploy the site as well subsequent commits (to this directory). Can we set this up with GitHub Pages?

**Comment by loteoo** [comment]
I'm not very experienced with GitHub Pages and don't know of a way to automate this, but basically, running `npm run build-site` in the root directory will build the site and output the files in the `docs` folder.

One way it could work for sure is to run it from a local development machine and commit the minified assets. Unfortunately this would pollute the repo a little bit but it's a way it would work directly.

Maybe some fancy GitHub Actions script could build the files in a `docs` branch, and that would be deployed?

Anyone here knows a way we could deploy this without polluting the repo?

**Comment by sergey-shpak** [comment]
@jorgebucaran @loteoo this can be done with github actions, how can I commit/add change to this branch?

**Comment by jorgebucaran** [comment]
@loteoo @sergey-shpak https://hyperapp.dev

**Pull request merged**